groupsforsections.py

```diff
@@ -2,69 +2,152 @@
+        sys.exit(1)
 
-    return response
+    check_the_response_state(response)
 
+    section_list = json.loads(response.text)
 
-def error_msg_handler(response):
-    # using yaml instead of json.load to get the error message as logs  will show non unicode values in logs
-    # for eg., [{u'message': u'Invalid access token.'}] ---> [{'message': 'Invalid access token.'}]
+    if not section_list:
+        logging.error('The course %s don\'t have any sections' % course_id)
+        sys.exit(1)
+
+    for section in section_list:
+        sections_dict[section['id']] = section['name']
+
+    next_page_url = groups_for_section.get_next_page_url(response)
+    if next_page_url is not None:
+        get_sections_for_course(sections_dict, groups_for_section, course_id, next_page_url)
+
+    return sections_dict
+
+
+def get_users_in_section(groups_for_section, users_in_section, section_id=None, next_page_url=None):
+    """
+    get all the users in the sections, paginates, parse Json,  exception handling.
+
+    :param GroupsForSections groups_for_section:
+    :param List users_in_section:
+    :param String section_id:
+    :param String next_page_url:
+    :return: List users[id]
+    """
+    logging.debug(get_users_in_section.__name__ + '() called')
+    try:
+        response = groups_for_section.users_in_section(section_id, next_page_url)
+
+    except requests.exceptions.RequestException as e:
+        logging.exception('getting USERS list has erroneous response ' + e.message)
+        sys.exit(1)
+
+    check_the_response_state(response)
+
+    user_list = json.loads(response.text)
+
+    for user in user_list:
+        users_in_section.append(user['user_id'])
 
-    error_res = yaml.safe_load(response.text)
-    logging.error('Api response has some errors: ' + str(error_res['errors']) + ' and Response Code: ' + str(
-            response.status_code))
+    next_page_url = groups_for_section.get_next_page_url(response)
+    if next_page_url is not None:
+        get_users_in_section(groups_for_section, users_in_section, None, next_page_url)
+
+    return users_in_section
+
+
+def create_group(groups_for_section, group_category_id, group_name, course_id):
+    """
+    creates a group with name same as section name, parses Json, do exception handling
+    :param GroupsForSections groups_for_section:
+    :param String group_category_id:
+    :param String group_name:
+    :return: String group_id
+    """
+    logging.debug(create_group.__name__ + '() called')
+    try:
+        response = groups_for_section.create_group(group_category_id, group_name, course_id)
+    except requests.exceptions.RequestException as e:
+        logging.exception('creating a GROUP has erroneous response ' + e.message)
+        sys.exit(1)
+
+    check_the_response_state(response)
+
+    group = json.loads(response.text)
+    return str(group['id'])
+
+
+def check_the_response_state(response):
+    if response.status_code != requests.codes.ok:
```

Comment:
NO, i am just checking if the response is 200 ok.  `requests.codes.ok= 200` . Changed the name of the function.